lib/hanami/configuration.rb

```diff
@@ -42,7 +43,7 @@ def initialize(&blk)
     #     # ...
     #   end
     def mount(app, options)
-      mounted[app] = App.new(app, options.fetch(:at))
+      mounted[app] = App.new(app, options.fetch(:at), options.fetch(:host) { nil })
```

Comment:
@artofhuman Good point, I simplified the read access for the host option. Actually we got quite confused with the existing documentation http://hanamirb.org/guides/routing/overview/#mounting-on-a-subdomain. It implies, that there is already the possibility to mount an application with a given host. Prior to my pull-request the host options was ignored and not delegated to hanami/router.